### PR TITLE
Disable CSI drivers installation on cluster where it's already turned on

### LINKED ISSUE #12008: Option to disable CSI drivers
### Description of the feature you would like to add / User story

As an admin, I would like to have the option to disable CSI drivers. 

Customers might want to bring in their own storage solutions; managed or self-hosted. In that case, deploying the CSI driver is redundant.

### PR DESCRIPTION
**Pull request by akash-gautam**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #12008

**What type of PR is this?**

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Disabled CSI addon on user clusters where it was enabled & then disabled using `DisableCSIDriver` option. The CSI addon is removed only if the CSI drivers created by it are not in use.
```

**Documentation**:
```
TBD
```

### CODE REVIEW

File: pkg/controller/seed-controller-manager/addon/addon_controller.go [hunk 1]
```diff
@@ -533,6 +534,62 @@ func (r *Reconciler) migrateHetznerCSIDriver(ctx context.Context, log *zap.Sugar
 	return nil
 }
 
+func (r *Reconciler) unInstallCSIDriver(ctx context.Context, log *zap.SugaredLogger, addon *kubermaticv1.Addon, cluster *kubermaticv1.Cluster) error {
+	userClusterClient, err := r.KubeconfigProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get client for usercluster: %w", err)
+	}
+	// Increased the timeout, some of the operations were getting timed out.
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```
Comment (ahmedwaleedmalik) [inline pkg/controller/seed-controller-manager/addon/addon_controller.go]: Please don't introduce new contexts and use the one already propagated.

```suggestion
	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
```

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

**Comment by ahmedwaleedmalik** [comment]
We need to discuss this PR first and then I'll do a proper review. I just want clarity on a few things.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by ahmedwaleedmalik** [comment]
@akash-gautam can we please shorten the release note a bit and make it a bit more concise?

**Comment by akash-gautam** [comment]
@ahmedwaleedmalik Updated the release notes, thanks.

**Comment by ahmedwaleedmalik** [comment]
Please add description in `What this PR does / why we need it` section for this PR.

**Pull request merged**